CRM/Core/BAO/Setting.php

```diff
@@ -178,6 +178,9 @@ public static function setItem(
     /** @var \Civi\Core\SettingsManager $manager */
     $manager = \Civi::service('settings_manager');
     $settings = ($contactID === NULL) ? $manager->getBagByDomain($domainID) : $manager->getBagByContact($domainID, $contactID);
+    if (self::isEnvironmentSet($name)) {
+      throw new api_Exception('CiviCRM Environment already set in civicrm.settings.php!');
```

Comment:
Trying to set it to same state in civicrm.settings.php should not raise an error. Trying to set its value to something different should be an error.

With regard to how error is raised/returned, I expect that it should be a friendly error message in UI, and a call to the REST API should return error=1 with error message somehow.